### PR TITLE
pkg/nanors: add reed solomon codec implementation

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

[nanors](https://github.com/sleepybishop/nanors) provides fast reed solomon codes in GF2^8.
On Cortex-M4 the implementation (`rs.o`) only consumes a little less than 2k ROM. 

### Testing procedure

A test application is provided in `tests/pkg_nanors`.
It generates random data, parity shards and then deletes random shards and attempts to reconstruct them.

### CODE REVIEW

**Comment by kaspar030** [comment]
Nice, stack usage is quite low:
```
        pid | name                 | state    Q | pri | stack  ( used) ( free) | base addr  | current     
          - | isr_stack            | -        - |   - |    512 (  164) (  348) | 0x20000000 | 0x200001c8
          1 | main                 | running  Q |   7 |   1536 (  504) ( 1032) | 0x20000f98 | 0x2000151c 
            | SUM                  |            |     |   2048 (  668) ( 1380)
```
(on nrf52840dk, probably the printf of ps used more)

**Review by kaspar030 — APPROVED** [review]
ACK. Nice library!

File: tests/pkg_nanors/tests/01-run.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python3
+
+import sys
+from testrunner import run
+
+
+def testfunc(child):
+    child.expect('START')
+    child.expect('SUCCESS')
```
Comment (aabadie) [inline tests/pkg_nanors/tests/01-run.py:9]:
```suggestion
    child.expect_exact('START')
    child.expect_exact('SUCCESS')
```

File: tests/pkg_nanors/main.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+    puts("START");
+    reed_solomon_init();
+    rs_t *rs = reed_solomon_new(DATA_SHARDS, RECOVERY_SHARDS);
+    if (!rs) {
+        puts("failed to init codec");
+        return -1;
+    }
+
+    printf("total: %d shards (%d bytes)\n", ARRAY_SIZE(marks), sizeof(data));
```
Comment (kaspar030) [inline tests/pkg_nanors/main.c:61]:
```
 main.c: In function â€˜mainâ€™:
main.c:61:21: error: format â€˜%dâ€™ expects argument of type â€˜intâ€™, but argument 2 has type â€˜long unsigned intâ€™ [-Werror=format=]
   61 |     printf("total: %d shards (%d bytes)\n", ARRAY_SIZE(marks), sizeof(data));
      |                    ~^
      |                     |
      |                     int
      |                    %ld
```

6lowpan-clicker...

Comment (kaspar030) [reply tests/pkg_nanors/main.c:61]: and pic32-wifire

**Comment by kaspar030** [comment]
Apart from the mips type mismatch (how annoying, didn't we deprecate mips...), `nucleo-l011k4 arduino-duemilanove stk3200 arduino-uno arduino-nano nucleo-f031k6 atmega328p-xplained-mini arduino-leonardo samd10-xmini stm32f030f4-demo atmega328p` are rom overflows.

**Pull request merged**